auth_helper.py

```diff
@@ -0,0 +1,68 @@
+import os
+import pathlib
+
+import json5
+import sys
+import requests
+import msal
+
+class AuthHelper:
+    tenantId = os.environ["tenant_id"]
+    scope = os.environ["scope"]
+    authority = os.environ["authority"]
+    appId = os.environ["appId"]
+    accessToken = ""
+    def __init__(self,**kwargs):
+        for key,value in kwargs.items():
+            setattr(self,key,value)
+
+    def adoAuthenticate(self):
+        mscache = msal.SerializableTokenCache()
+        output = pathlib.Path(__file__).parent / pathlib.Path("token.bin")
+        if os.path.exists(output):
+            print("Deserializing cached credentials.")
+            mscache.deserialize(open(output, "r").read())
+            
+        app = msal.PublicClientApplication(
+            client_id=self.appId,
+            token_cache=mscache
+        )
+        accounts = app.get_accounts()
+        if accounts:
+            result = app.acquire_token_silent(
+                scopes=[self.scope],
+                account=accounts[0]
+            )
+            if mscache.has_state_changed:
+                with open(output, "w") as cache_file:
+                    print("Caching credentials.")
+                    cache_file.write(mscache.serialize())
+            if result is not None:
+                if "access_token" in result:
+                    print("Found access token.")
+                    return result
+                else:
+                    raise RuntimeError(result.get("error_description"))
```

Comment:
This should be a custom error type.

Reply:
And provide an error message for the case where `error_description` isn't populated.